libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -236,6 +238,19 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     }
   }
 
+  private listenForIonModalWillPresent() {
+    if (!this.ionModalElement) return;
+
+    if (this.config.interactWithBackground) {
```

Comment:
Is it feasible to reduce this check so we always add the class if `this.config.interActWithBackground` is set? Then I think we will get minimum flicker, and the animation of the drawer will work.
So, e.g.:

```
if (this.config.interactWithBackground) {
      this.renderer.addClass(
        this.windowRef.document.body,
        this.ALLOW_BACKGROUND_SCROLL_CLASS_NAME
      );
    }
```

Reply:
I think an even better way to solve this would be to add this functionality to `modal.helper.ts` which can add the class to the body element before even opening the modal - that is right before calling `ionicModal.present`
@bdytos Ok?

Reply:
Possible fix in current commit